agent/app/agent_unix.go

```diff
@@ -0,0 +1,77 @@
+// +build linux
```

Comment:
I want it to be in the `sighandler` package, because the `SIGUSR1` and `SIGTERM` are already been handled there. It would be good to have all the signal handlers in one package. But I'm also fine with putting it here for now, since it's only needed when the `task networking` feature is needed.